packages/compose/src/hooks/use-dialog/README.md

```diff
@@ -0,0 +1,44 @@
+`useDialog`
+===========
+
+React hook to be used on a dialog wrapper to enable the following behaviors:
+
+ - constrained tabbing.
+ - focus on mount.
+ - return focus on unmount.
+ - focus outside.
+
+## Returned value
+
+The hooks returns an array composed of the two following values:
+
+### `ref`
+
+- Type: `Object|Function`
+
+A React ref that must be passed to the DOM element where the behavior should be attached.
+
+### `props`
+
+- Type: `Object`
+
+Extra props to apply to the wrapper.
+
+## Usage
+
+```jsx
+import { __experimentalUseDialog as useDialog } from '@wordpress/compose';
+
+const MyDialog = () => {
+	const [ ref, extraProps ] = useDialog( { 
```

Comment:
Why is ref separate? Ref is also a prop that can be spread?

Reply:
That's still subject to change, my hope is that we can remove the props entirely and just return the ref. more explanations here https://github.com/WordPress/gutenberg/pull/27643#issuecomment-743209335